Automattic/woocommerce-payments
# Pass product, blocks cart, classic cart and checkout on get WooPay session

Fixes https://github.com/Automattic/woopay/issues/2804

#### Changes proposed in this Pull Request
This PR extends from the [pass Blocks checkout appearance on init WooPay](https://github.com/Automattic/woocommerce-payments/pull/9075) PR and use `getAppearanceType` to get element selectors from each page since a shopper doesn't always visit the checkout page before going to the WooPay checkout page.

EPIC https://github.com/Automattic/woopay/issues/2781
SPIKE [here](https://woopayp2.wordpress.com/2024/07/01/spike-global-themes-support/#proof-of-concept-for-option-3-preferred)
PT [here](https://woopayp2.wordpress.com/2024/07/11/project-thread-woopay-global-themes-support/)
[Related WooPay PR](https://github.com/Automattic/woopay/pull/2805)


https://github.com/user-attachments/assets/6e3c6d5a-bfad-4af6-a629-f7ff867f3965


#### Testing instructions
* Test with the latest WooPay trunk
* Go to WooPay admin -> WCPay Dev and enable `WooPay global theme support` feature flag
* Update the merchant styles:
1. Install and activate a theme that supports the site editor from admin -> appearance -> theme
1. Update typography by visiting Theme -> Editor -> Styles
OR
1. Update merchant styles ( ex. background color, text color, border styles, etc ) without using [theme that supports the site editor](https://wordpress.com/themes/filter/full-site-editing) from wp-admin appearance


Test 1 ( The latest WooPayments version ):
1. Test this PR with the latest WooPayments version that includes `getAppearance` on the merchant site
1. Open dev tools
2. As an existing WooPay shopper, go to the Blocks checkout page and initialize WooPay with the **email field redirect**
3. Find the `get_woopay_session` request from dev tools and see `appearance` is passing with the request
4. Should see WooPay support background color, color, and font family
5. Test `Return to cart` -> Blocks checkout page -> WooPay button
6. Should see WooPay support background color, color, and font family
7. Test `Checkout as guest` and initialize WooPay as a new WooPay shopper with the **email field redirect**
8. Should see WooPay support background color, color, and font family
9. Test `Return to cart` -> Blocks checkout page
10. As a new WooPay shopper, initialize WooPay with the WooPay button
11. Should see WooPay support background color, color, and font family
12. Repeat the above by initializing WooPay on the product page, Blocks cart page, Classic cart page, and Classic checkout page.

Test 2 ( An older WooPayments version ):

1. Test an **older version** of WooPayments like [`7.6.0`](https://github.com/Automattic/woocommerce-payments/releases/download/7.6.0/woocommerce-payments.zip) that doesn't have `getAppearance` on the merchant site for **regression** test
2. Repeat above testing instructions

-------------------


**Post merge**
